txnsync/outgoing_test.go

```diff
@@ -0,0 +1,381 @@
+
+	err := asyncEncoder.asyncEncodeAndSend(nil)
+
+	a.Nil(err)
+	a.False(warnCalled)
+	a.True(sendPeerMessageCalled)
+	a.Nil(asyncEncoder.messageData.transactionGroups)
+}
+
+// TestAsyncEncodeAndSendErr Tests response when encodeTransactionGroups returns an error
+func TestAsyncEncodeAndSendErr(t *testing.T) {
+	a := require.New(t)
+
+	var s syncState
+	s.clock = timers.MakeMonotonicClock(time.Now())
+	warnCalled := false
+	s.log = mockAsyncLogger{warnCalled: &warnCalled}
+	sendPeerMessageCalled := false
+	s.node = mockAsyncNodeConnector{called: &sendPeerMessageCalled}
+	s.messageSendWaitGroup = sync.WaitGroup{}
+
+	txnGrps := []transactions.SignedTxGroup{
+		transactions.SignedTxGroup{
+			Transactions: []transactions.SignedTxn{
+				transactions.SignedTxn{
+					Txn: transactions.Transaction{
+						Type: protocol.UnknownTx,
+					},
+				},
+			},
+		},
+	}
+
+	asyncEncoder := messageAsyncEncoder{
+		state: &s,
+		messageData: sentMessageMetadata{
+			message: &transactionBlockMessage{
+				Version: txnBlockMessageVersion,
+				Round:   1,
+			},
+			transactionGroups: txnGrps,
+			peer:              &Peer{},
+		},
+		roundClock: timers.MakeMonotonicClock(time.Now()),
+	}
+
+	asyncEncoder.state.messageSendWaitGroup.Add(1)
+
+	err := asyncEncoder.asyncEncodeAndSend(nil)
+
+	a.Nil(err)
+	a.True(warnCalled)
+	a.True(sendPeerMessageCalled)
+
+}
+
+// TestAsyncEncodeAndSend Tests that SendPeerMessage is called in the node connector
+func TestAsyncEncodeAndSend(t *testing.T) {
+	a := require.New(t)
+
+	var s syncState
+	s.clock = timers.MakeMonotonicClock(time.Now())
+	s.log = mockAsyncLogger{}
+	sendPeerMessageCalled := false
+	s.node = mockAsyncNodeConnector{called: &sendPeerMessageCalled}
+	s.messageSendWaitGroup = sync.WaitGroup{}
+
+	asyncEncoder := messageAsyncEncoder{
+		state: &s,
+		messageData: sentMessageMetadata{
+			message: &transactionBlockMessage{
+				Version: txnBlockMessageVersion,
+				Round:   1,
+			},
+			peer: &Peer{},
+		},
+		roundClock: timers.MakeMonotonicClock(time.Now()),
+	}
+
+	asyncEncoder.state.messageSendWaitGroup.Add(1)
+
+	err := asyncEncoder.asyncEncodeAndSend(nil)
+
+	a.Nil(err)
+	a.True(sendPeerMessageCalled)
+
+}
+
+// TestAssemblePeerMessage1 Tests assemblePeerMessage with messageConstBloomFilter msgOps
+func TestAssemblePeerMessage1(t *testing.T) {
+	a := require.New(t)
+
+	s := syncState{clock: timers.MakeMonotonicClock(time.Now())}
+
+	s.profiler = makeProfiler(1*time.Millisecond, s.clock, s.log, 1*time.Millisecond)
+
+	peer := Peer{}
+
+	pendingTransactions := pendingTransactionGroupsSnapshot{
+		pendingTransactionsGroups: []transactions.SignedTxGroup{
+			transactions.SignedTxGroup{},
+		},
+	}
+
+	peer.setLocalRequestParams(111, 222)
+	peer.lastReceivedMessageTimestamp = 100
+	peer.lastReceivedMessageLocalRound = s.round
+
+	expectedFilter := s.makeBloomFilter(requestParams{Offset: 111, Modulator: 222}, pendingTransactions.pendingTransactionsGroups, &s.lastBloomFilter)
+
+	s.isRelay = true
```

Comment:
You are only testing this for the relay case. 
assemblePeerMessage for messageConstBloomFilter has different code path if it is not a relay.

Reply:
Resolved via #2654